File: services/education/internal/model/school_holiday.go
package model

type SchoolHolidayVO struct {
	ID        int64  `json:"id"`
	InstID    int64  `json:"instId"`
	Name      string `json:"name"`
	StartDate string `json:"startDate"`
	EndDate   string `json:"endDate"`
	Source    string `json:"source"`
	Sort      int    `json:"sort"`
}

type SchoolHolidayMutation struct {
	ID        *int64 `json:"id,omitempty"`
	Name      string `json:"name"`
	StartDate string `json:"startDate"`
	EndDate   string `json:"endDate"`
	Source    string `json:"source,omitempty"`
	Sort      *int   `json:"sort,omitempty"`
}

type SchoolHolidayDeleteDTO struct {
	ID *int64 `json:"id,omitempty"`
}

File: institution-admin/src/api/business-settings/school-holiday.ts
import { useGet, usePost } from '~/utils/request'

export interface SchoolHolidayItem {
  id: number
  instId?: number
  name: string
  startDate: string
  endDate: string
  source: 'statutory' | 'custom'
  sort?: number
}

export interface SchoolHolidayMutation {
  id?: number
  name: string
  startDate: string
  endDate: string
  source?: 'statutory' | 'custom'
  sort?: number
}

export function listSchoolHolidaysApi() {
  return useGet<SchoolHolidayItem[]>('/api/v1/school-holidays')
}

export function saveSchoolHolidayApi(data: SchoolHolidayMutation) {
  return usePost<{ id: number }>('/api/v1/school-holidays/save', data)
}

export function deleteSchoolHolidayApi(data: { id: number }) {
  return usePost<boolean>('/api/v1/school-holidays/delete', data)
}

export function resetSchoolHolidaysApi() {
  return usePost<boolean>('/api/v1/school-holidays/reset', {})
}
